File: ai_recommend/infrastructure/db/surreal_db.py
from ai_recommend.infrastructure.db.config import DatabaseConfig
from surrealdb import Surreal

class SurrealDb:
    def __init__(self, config: DatabaseConfig):
        self.config = config
        self.db = Surreal(config.url)
        self.db.signin({"username": self.config.username, "password": self.config.password})
        self.db.use(self.config.namespace, self.config.database)

File: users_api/src/infrastructure/surrealdb.rs
use surrealdb::engine::remote::ws::{Client, Ws};
use surrealdb::opt::auth::Root;
use surrealdb::{Result, Surreal};

pub struct DbConfig {
    pub host: String,
    pub port: u16,
    pub username: String,
    pub password: String,
    pub namespace: String,
    pub database: String,
}

impl DbConfig {
    fn get_address(&self) -> String {
        format!("{}:{}", self.host, self.port)
    }
}

pub async fn get_database(config: DbConfig) -> Result<Surreal<Client>> {
    let db: Surreal<Client> = Surreal::init();
    let _ = db.connect::<Ws>(config.get_address()).await?;
    let _ = db.signin(Root {
        username: &config.username,
        password: &config.password,
    })
    .await?;
    let _ = db.use_ns(config.namespace).use_db(config.database).await?;
    Ok(db)
}
